spec/session_spec.rb

```diff
@@ -41,4 +44,28 @@ class Session < Binance::Session
       expect(subject.timeout).to eq(nil)
     end
   end
+
+  context 'should init with proxy' do
+    before do
+      allow(Faraday).to receive(:new).with(url: 'https://api.binance.com', proxy: 'proxy_url')
+    end
+
+    subject { Session.new(proxy_url: 'proxy_url' ) }
+    it 'include the proxy options' do
+      subject.send(:build_connection)
+      expect(Faraday).to have_received(:new).with(url: 'https://api.binance.com', proxy: 'proxy_url')
+    end
+  end
+
+  context 'should init without proxy' do
```

Comment:
se debería usa `when`: when no proxy_url option is given

aunque veo que en otro contexto que había antes usan should, que es raro, should es la nomeclatura antigua para ejemplos.